platforms/core-configuration/model-core/src/integTest/groovy/org/gradle/api/provider/GroovyPropertyAssignmentIntegrationTest.groovy

```diff
@@ -44,16 +44,19 @@ class GroovyPropertyAssignmentIntegrationTest extends AbstractProviderOperatorIn
         groovyBuildFile(inputDeclaration, inputValue, "=")
 
         expect:
+        if (inputType == "Property<MyEnum>") {
```

Comment:
Could you just do this as a separate test method vs conditionally checking for something here?